src/test/s3/util/fileViewerManager.test.ts

```diff
@@ -3,6 +3,135 @@
+    let testCache: Set<string>
+    let tempPath: string
+    let parent: S3BucketNode | S3FolderNode
+    let commands: FakeCommands
+    let tempFile: vscode.Uri
+
+    beforeEach(async function () {
+        window = mock()
+        s3 = mock()
+        parent = mock()
+        tempPath = await makeTemporaryToolkitFolder()
+        testCache = new Set<string>()
+        commands = new FakeCommands()
+        fileViewerManager = new S3FileViewerManager(testCache, instance(window), commands, tempPath)
+        file = new DefaultFile({
+            partitionId: 'aws',
+            bucketName,
+            key,
+            lastModified: creationDate,
+            sizeBytes,
+        })
+        testNode = new S3FileNode({} as any, file, instance(parent), instance(s3))
+
+        let completePath = getStringHash(readablePath(testNode))
+        //let completePath = readablePath(testNode).slice(4).split('/').join('%')
+        completePath = path.join(tempPath, 'S3%' + completePath)
+        tempFile = vscode.Uri.file(completePath)
+        testutil.toFile('bogus', tempFile.fsPath)
+    })
+
+    describe('retrieves file from s3 if not in temp or invalid date', function () {
+        this.beforeEach(function () {})
+
+        it('prompts if file has no specified size', async function () {
+            when(window.showInformationMessage(anything(), anything(), anything())).thenReturn(
+                Promise.resolve('Cancel')
+            )
+
+            file = new DefaultFile({
+                partitionId: 'aws',
+                bucketName,
+                key,
+                lastModified: creationDate,
+                sizeBytes: undefined,
+            })
+            testNode = new S3FileNode({} as any, file, {} as any, instance(s3))
+
+            assert.strictEqual(await fileViewerManager.getFile(testNode), undefined)
+        })
+
+        it('prompts if file size is greater than 4MB', async function () {
+            when(window.showInformationMessage(anything(), anything(), anything())).thenReturn(
+                Promise.resolve('Cancel')
+            )
+
+            file = new DefaultFile({
+                partitionId: 'aws',
+                bucketName,
+                key: 'hello',
+                lastModified: creationDate,
+                sizeBytes: 5 * Math.pow(10, 6),
+            })
+
+            testNode = new S3FileNode({} as any, file, {} as any, instance(s3))
+
+            assert.strictEqual(await fileViewerManager.getFile(testNode), undefined)
+        })
+
+        it('downloads and adds to cache', async function () {
+            when(s3.downloadFile(anything())).thenResolve()
+            when(parent.getChildren()).thenResolve([testNode])
+
+            assert.ok(!testCache.has(testNode.file.arn))
+
+            await fileViewerManager.getFile(testNode)
+
+            assert.ok(testCache.has(testNode.file.arn))
+        })
+    })
+
+    describe('uses cache', function () {
```

Comment:
Add a test that shows that the cache is used for unchanged cached >4MB files?

Reply:
It is used normally, I have the test that `prompts user if file is greater than 4MB` if the prompt is confirmed it should continue with the regular flow. Do you think it makes sense to add another test for this though?

Reply:
If we're thinking of these tests as 'blackboxes' then it doesn't hurt to add the test. This becomes relevant if your caching/prompting logic changes in a significant way.